src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/repository/LicenceRepository.kt

```diff
@@ -36,6 +36,22 @@ interface LicenceRepository : JpaRepository<Licence, Long>, JpaSpecificationExec
     """,
   )
   fun getApprovedLicencesOnOrPassedReleaseDate(): List<Licence>
+
+  @Query(
+    """
+    SELECT l
+        FROM Licence l 
+        WHERE (l.licenceExpiryDate < CURRENT_DATE AND l.topupSupervisionExpiryDate >= CURRENT_DATE 
+                AND l.typeCode IN (uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceType.AP_PSS))
+        AND l.statusCode IN (
+            uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.VARIATION_IN_PROGRESS,
+            uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.VARIATION_SUBMITTED,
+            uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.VARIATION_REJECTED,
+            uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.VARIATION_APPROVED
+        )
+    """,
+  )
+  fun getAllVariedLicencesInPSSPeriod(): List<Licence>
```

Comment:
I think this is ok with out optimising - it does do multiple n*1 issues but it needs them for the delete and I suspect the number of licences that this will affect to be quite small. 

I think we'll need to observe this in preprod/prod

Reply:
yes, we can observe the performance in preprod/prod and then we can refactor if needed.